apps/user-profile/src/app/components/header/header.component.ts

```diff
@@ -27,14 +31,42 @@ export class HeaderComponent implements OnInit {
 
   constructor( private storeService: StoreService,
                private sanitizer: DomSanitizer,
-               private translate: TranslateService) { }
+               private translate: TranslateService,
+               private utilsService: UtilsService,
+               private authzResolverService: AuthzResolverService,
+               private authService: AuthService,) { }
   logo: any;
+  adminGuiUrl: string;
+  linkRoles: string[];
+  activeLink = false;
 
   ngOnInit() {
     this.translate.onLangChange.subscribe(lang => {
       this.label = this.storeService.get(`header_label_${lang.lang}`)
+      this.adminLabel = this.storeService.get(`admin_gui_label_${lang.lang}`);
     })
     this.logo = this.sanitizer.bypassSecurityTrustHtml(this.storeService.get('logo'));
+
+    this.linkRoles = this.storeService.get('link_to_admin_gui_by_roles');
+    for (const roleKey in this.storeService.getPerunPrincipal().roles) {
+      if (this.linkRoles.includes(roleKey)) {
+        this.activeLink = true;
+      }
+    }
+
+    if (this.activeLink) {
+      this.utilsService.getAppsConfig().subscribe(config => {
+        this.adminGuiUrl = !!config.brands[0].newApps.admin ? config.brands[0].newApps.admin : null;
+      });
+    }
```

Comment:
move this code to separate function

Reply:
This block of code has been moved to the separate function.